### ISSUE TITLE
Behavior of HttpObjectAggregator with too large request

### ISSUE THREAD

**Issue by jorcasso** [netty/netty#9153]
I'm implementing a HTTP server with netty 4.1.33, the pipeline looks like this:
```
pipeline.addLast("http-request-decoder", new HttpRequestDecoder());
pipeline.addLast("http-response-encoder", new HttpResponseEncoder());
pipeline.addLast("http-request-aggregator", new HttpObjectAggregator(maxContentLength));
pipeline.addLast("handler", new MyRequestHandler());
```

I have observed that when a request arrives with a larger `Content-Length` than configured in `HttpObjectAggregator`, the aggregator itself returns a 413 error response and `MyRequestHandler` receives a `FullHttpRequest` with unsuccessful decoder result, the error in the decoder result is:

```
io.netty.handler.codec.TooLongFrameException: An HTTP line is larger than 4096 bytes.
```

That error is weird to me and I found that it is caused by the `HttpRequestDecoder`, it starts to decode a new HTTP request using the body of the request that failed with the 413. Is that behavior correct? shouldn't the body be discarded when the `HttpObjectAggregator` returns the 413 error? or should I handle it by myself?

Looking into the source code it seems this piece of code in `HttpObjectAggregator` is causing this behavior:

```
// If an oversized request was handled properly and the connection is still alive
// (i.e. rejected 100-continue). the decoder should prepare to handle a new message.
HttpObjectDecoder decoder = ctx.pipeline().get(HttpObjectDecoder.class);
if (decoder != null) {
    decoder.reset();
}
```

**Comment by doom369** [comment]
@jorcasso this is no longer in case with https://github.com/netty/netty/pull/9017/files. Please try to switch to the latest netty version.

**Comment by jorcasso** [comment]
Thanks for answering @doom369. Sorry, I should have checked first the last version of netty. However, it seems that pull has not been released, as I can reproduce the same behavior with version 4.1.36, and in the source code I can not find those changes. It's like that pull request was dropped for some reason.

**Comment by normanmaurer** [comment]
@jorcasso you are right we somehow missed to cherry-pick it into 4.1 :( ... Let me do it

**Comment by normanmaurer** [comment]
Here we go https://github.com/netty/netty/pull/9156

**Comment by normanmaurer** [comment]
I just merged the change into 4.1

**Comment by normanmaurer** [comment]
So should be fixed in 4.1.37.Final once released

**Comment by jorcasso** [comment]
Thanks @normanmaurer
